libraries/chez/common/immutable/list.effekt

```diff
@@ -22,6 +22,45 @@ def foreach[A](l: List[A]) { f: A => Unit } : Unit =
       foreach(rest) { a => f(a) }
   }
 
+def foreachIndex[A](l: List[A]) { f: (Int, A) => Unit } : Unit = {
```

Comment:
Most other languages that I am aware of first list the element, then the index. WDYT?

Reply:
No preference from my point of view, I just followed the order which was already present in `immutable/list` on the JS backend 😇 

https://github.com/effekt-lang/effekt/blob/master/libraries/js/immutable/list.effekt#L23

Reply:
well... what did I think back then? Let's merge first, then share, then reorganize.

Reply:
> Most other languages that I am aware of first list the element, then the index. WDYT?

I took a quick look at other languages:

Python with `enumerate`:
```py
for index, elem in enumerate(myIterator):
```

Go with `range`:
```
for index, elem := range mySlice { ... }
```

C++20 with `enumerate`:
```
for (auto const [index, elem] : std::views::enumerate(v))
```

Kotlin:
```
fun <T> Iterable<T>.forEachIndexed(
    action: (index: Int, T) -> Unit)
)
```

Haskell (`vector` library):
```
iforM :: Monad m => Vector a -> (Int -> a -> m b) -> m (Vector b)
```

... but Scala uses `zipWithIndex` and has the `(T, Int)` order instead

Reply:
Funny, I looked at JS, Scala, Ruby...